### DISCUSSION TITLE
Promoting children if they don't match parent containers

### DISCUSSION THREAD

**Discussion by nath1as** [piroor/treestyletab#3005] [Q&A] [answered]
I'm using TST with containers and would like to make typed trees, so that parent could only have a child if it was the same type otherwise the new tab would be a sibling. 

Is this possible to do?

**Comment by irvinm** [comment, marked as answer]
I'm pretty sure you can't do this directly with TST.  I was thinking about how you might do this with CSS, but am not sure it is possible.  I'm sure this could be done with a helper addon.  (You would just need to watch for new tabs, compare container information [if available for tab information], change the parent-child relationship based on the comparison result)

**Comment by nath1as** [comment]
Yea, for now,my solution is opening all in new tabs and indenting with keyboard, 
maybe there is some container extensions that would organize tabs like that ...

**Comment by piroor** [comment]
This functionality looks beyond the project scope of TST, instead I recommend you to develop a new helper addon for TST. It will need to watch two events: new tabs opened with container (observable via `tabs.onCreated` handler)  and tree structure changes on TST.
https://github.com/piroor/treestyletab/wiki/API-for-other-addons#when-tree-structure-is-modified
